app/routes/posts/new.js

```diff
@@ -14,21 +14,22 @@ export default Ember.Route.extend({
         imageUrl: newPost.imageUrl,
         fileName: newPost.fileName,
         caption: newPost.caption,
+        file: newPost.file,
         user: user
       });
 
       if (newPost.file !== null) {
         newPost.file.formData = {
           data: JSON.stringify(postCreation.serialize().data)
         }
-        newPost.file.submit();
+        newPost.file.submit().then(function(data){
```

Comment:
If you are not gonna do anything in the success callback, no need to have it. Same below.